core/src/storage2/collections/binary_heap/children.rs

```diff
@@ -96,16 +96,20 @@ where
         Self { left, right }
     }
 
-    /// Returns `true` if no child exists in this object, otherwise `false`.
-    pub fn is_empty(&self) -> bool {
-        self.left.is_none() && self.right.is_none()
+    /// Returns the number of existent children in this object.
+    pub fn count(&self) -> usize {
+        CHILDREN_PER_NODE as usize
+            - [self.left.as_ref(), self.right.as_ref()]
+                .iter()
+                .flatten()
+                .count()
```

Comment:
Could you benchmark this against:
`self.left.is_some() as usize + self.right.is_some() as usize`
?

Reply:
I benched it and your version was faster (took about 6% less time).